any-sketch/src/test/cc/any_sketch/crypto/sketch_encrypter_test.cc

```diff
@@ -404,5 +404,35 @@ TEST_F(SketchEncrypterTest, CombineElGamalPublicKeysInvalidInputShouldThrow) {
               StatusIs(absl::StatusCode::kInvalidArgument, "Invalid ECPoint"));
 }
 
+TEST_F(SketchEncrypterTest, NoisesShouldHaveTheSameIndex) {
+  Context ctx;
+  ASSERT_OK_AND_ASSIGN(ECGroup ec_group, ECGroup::Create(kTestCurveId, &ctx));
+
+  std::string encrypted_sketch;
+  int values_per_register = 5;
+  int ciphertexts_per_register = (values_per_register + 1) * 2;
+
+  EncryptSketchRequest::PublisherNoiseParameter noise_parameter;
+  noise_parameter.set_epsilon(1);
+  noise_parameter.set_delta(0.1);
+  noise_parameter.set_publisher_count(3);
+
+  ASSERT_TRUE(sketch_encrypter_
+                  ->AppendNoiseRegisters(noise_parameter, values_per_register,
+                                         encrypted_sketch)
+                  .ok());
+
+  std::vector<std::string> cipher_words = GetCipherStrings(encrypted_sketch);
+  int noise_register_count = cipher_words.size() / ciphertexts_per_register;
+
+  ASSERT_GT(noise_register_count, 0);
+  for (int i = 0; i < noise_register_count; ++i) {
```

Comment:
i < cipher_words.size() won't make it look safe, since it calls cipher_words[i+1] later.

if we write i < cipher_words.size()-1, then the -1 is a little weird.

so I changed it to " i <= cipher_words.size()-ciphertexts_per_register", which may be easier to follow or maybe harder.